### PR TITLE
AB#2365 Upgrade k8s base deployments (add full support for k8s 1.25)

### PR DESCRIPTION
**Pull request by malt3**
### Proposed change(s)
- Add container image for CCM GCP v25.2.0
- Upgrade versions of kubernetes base components

### CODE REVIEW

File: internal/versions/versions.go [hunk 1]
```diff
@@ -111,13 +111,13 @@ var VersionConfigs = map[ValidK8sVersion]KubernetesVersion{
...
-		CloudControllerManagerImageGCP: "ghcr.io/edgelesssys/cloud-provider-gcp:v24@sha256:8ee4261980019d3ee8517e12f36fc313fe3ea3e44dd40ee2e004b57f6e5ef171",
+		CloudControllerManagerImageGCP: "ghcr.io/edgelesssys/cloud-provider-gcp:v24.0.0@sha256:8ee4261980019d3ee8517e12f36fc313fe3ea3e44dd40ee2e004b57f6e5ef171",
 		// CloudControllerManagerImageAzure is the CCM image used on Azure.
 		CloudControllerManagerImageAzure: "mcr.microsoft.com/oss/kubernetes/azure-cloud-controller-manager:v1.24.8@sha256:109de6e558fa3f68c5d786adddf2eb864fde36c1887a3a04e0d11c85a39954c0",
 		// CloudNodeManagerImageAzure is the cloud-node-manager image used on Azure.
 		CloudNodeManagerImageAzure: "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.24.8@sha256:b98d0a9a683d86c58ea27956f3c7e1263f95541e41b5053c6209c4c5fc647cba",
 		// External service image. Depends on k8s version.
-		ClusterAutoscalerImage: "k8s.gcr.io/autoscaling/cluster-autoscaler:v1.23.1@sha256:cd2101ba67f3d6ec719f7792d4bdaa3a50e1b716f3a9ccee8931086496c655b7",
+		ClusterAutoscalerImage: "k8s.gcr.io/autoscaling/cluster-autoscaler:v1.24.0@sha256:5bd22353ae7f30c9abfaa08189281367ef47ea1b3d09eb13eb26bd13de241e72",
```
Comment (katexochen) [inline internal/versions/versions.go:120]: Where did you find this image? No release/tag exists with `v1.24.0` at https://github.com/kubernetes/autoscaler/releases .

Comment (3u13r) [reply internal/versions/versions.go:120]: Exists on my lookup 😄 : https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.24.0

**Review by 3u13r — APPROVED** [review]
LGTM

File: internal/versions/versions.go (hunk 1 above)
Comment (malt3) [reply internal/versions/versions.go:120]: To extend on that: I use the curl commands from this guide to detect all tags for images on any container registry: https://github.com/edgelesssys/constellation/blob/main/.github/docs/upgrade-kubernetes.md

**Pull request merged**